### PR TITLE
define queue model function

### PR DESCRIPTION
**Pull request by PaulLockett**
defines the queueModelForTraining function on the ImageLearningService and its associated types.

### CODE REVIEW

File: src/core/services/images/LearningService.ts [hunk 1]
```diff
@@ -66,8 +67,26 @@ export class ImageLearningService
     return response.json();
   }
 
-  queueModelForTraining(id: string): Promise<void> {
-    throw new Error("Method not implemented.");
+  public async queueModelForTraining(id: string, payload:QueueImageModelForTrainingPayload): Promise<TrainingJobSchema> {
```
Comment (atbe) [inline src/core/services/images/LearningService.ts:70]: I just pushed a change to the linting workflow to autocheck formatting.

I suggest you install these two plugins (if you're using VSCode):
- https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint
- https://marketplace.visualstudio.com/items?itemName=rvest.vs-code-prettier-eslint

I just pushed a vscode workspace file that you can open that will automatically pick the extension so your code is formatter automatically as you save/paste so you don't have to think about it.

File: src/core/services/images/LearningService.ts [hunk 2]
```diff
@@ -3,8 +3,9 @@ import { ImageLearningServiceInterface } from "../../../interfaces/images/Learni
 import SdkOptions from "../../../types/SdkOptions";
 
 import LeapImageService from "../../../types/services/LeapImageService";
-import { LeapModelSchema } from "../../../types/schemas/Model";
+import { LeapModelSchema, TrainingJobSchema } from "../../../types/schemas/Model";
 import CreateImageModelPayload from "../../../types/schemas/CreateImageModelPayload";
+import QueueImageModelForTrainingPayload from "../../../types/schemas/QueueImageModelForTrainingPayload";
 import { FetchClient } from "../../util";
 
 export class ImageLearningService
```
Comment (atbe) [inline src/core/services/images/LearningService.ts:11]: Can you add a docstring throughout this file:

- The class
- The methods
- The parameters
- The return objects

File: src/interfaces/images/LearningServiceInterface.ts [hunk 3]
```diff
@@ -24,9 +25,10 @@ export interface ImageLearningServiceInterface {
    * Queues a model for training and. A new version is created. This is a billable action.
    *
    * @param id - The ID of the model to queue for training. (uuid)
+   * @param payload - the payload for training a model.
    * @returns - The training job corresponding to the new version.
    */
-  queueModelForTraining: (id: string) => Promise<void>;
+  queueModelForTraining: (id: string, payload:QueueImageModelForTrainingPayload) => Promise<TrainingJobSchema>;
```
Comment (atbe) [inline src/interfaces/images/LearningServiceInterface.ts:31]: Don't use any of the existing models - (if they're from commit `e413ef21f53f559ef6be898397421396790b1396` or earlier), that's legacy code (it'll be removed before release)

Here's an example of a data model class that utilizes the OpenAPI model and is ready to expose to consumers https://github.com/leap-api/Leap-Typescript-SDK/blob/master/src/types/schemas/remix/Remix.ts

For this interface, lets be sure to use newly written models that have proper ties to the OpenAPI models.

File: src/types/schemas/Model.ts
Comment (atbe) [inline src/types/schemas/Model.ts:1]: Let's avoid defining multiple types in one file - for the models related to the `Learning` or `FineTune` features, lets make a subfolder within the `schema` folder [here](https://github.com/leap-api/Leap-Typescript-SDK/tree/master/src/types/schemas) similar to the `remix` feature.

File: src/types/schemas/Model.ts [hunk 4]
```diff
@@ -4,3 +4,15 @@ export type LeapModelSchema = {
   subjectKeyword: string;
   subjectIdentifier: string;
 };
+
+export type TrainingJobSchema = {
```
Comment (atbe) [inline src/types/schemas/Model.ts:8]: Lets avoid redefining models, and instead extend the OpenAPI generated models

Here's an example of a data model class that utilizes the OpenAPI model and is ready to expose to consumers https://github.com/leap-api/Leap-Typescript-SDK/blob/master/src/types/schemas/remix/Remix.ts

File: src/types/schemas/QueueImageModelForTrainingPayload.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+/**
+ * The payload for  training an image model.
+ */
+export default interface QueueImageModelForTrainingPayload {
```
Comment (atbe) [inline src/types/schemas/QueueImageModelForTrainingPayload.ts:4]: Similar comment as above

**Review by atbe — CHANGES_REQUESTED** [review]

File: src/core/services/images/LearningService.ts (hunk 1 above)
Comment (PaulLockett) [reply src/core/services/images/LearningService.ts]: done

File: src/core/services/images/LearningService.ts (hunk 2 above)
Comment (PaulLockett) [reply src/core/services/images/LearningService.ts:15]: This has been done

File: src/interfaces/images/LearningServiceInterface.ts (hunk 3 above)
Comment (PaulLockett) [reply src/interfaces/images/LearningServiceInterface.ts]: Removed all usages

File: src/types/schemas/Model.ts
Comment (PaulLockett) [reply src/types/schemas/Model.ts:1]: all types are in separate files

File: src/types/schemas/Model.ts (hunk 4 above)
Comment (PaulLockett) [reply src/types/schemas/Model.ts]: Types are now extensions

File: src/types/schemas/QueueImageModelForTrainingPayload.ts (hunk 5 above)
Comment (PaulLockett) [reply src/types/schemas/QueueImageModelForTrainingPayload.ts]: done for his as well

**Pull request merged**